io/io/inc/ROOT/RIoUring.hxx

```diff
@@ -56,6 +61,91 @@ public:
       static const bool available = RIoUring::CheckIsAvailable();
       return available;
    }
+
+   /// Access the raw io_uring instance.
+   struct io_uring *GetRawRing() {
+      return &fRing;
+   }
+
+   /// Basic read event composed of RIOVec IO data and a target file descriptor.
+   struct RReadEvent {
+      RRawFile::RIOVec* fIoVec = nullptr;
+      int fFileDes = -1;
+   };
+
+   /// Submit a number of read events and wait for completion.
+   void SubmitReadsAndWait(std::vector<RReadEvent>& readEvents) {
+      auto numEvents = readEvents.size();
```

Comment:
Let's make batching part of it from the start, so that we easily can throw in many hundreds of requests

Reply:
Good idea, I tried to implement batching in 8c8a841 
We will see if the CI likes it!